extension/options/optionsView.jsx

```diff
@@ -498,3 +557,49 @@ const AboutSection = () => {
     </fieldset>
   );
 };
+
+const Tabs = ({
+  updateActiveTab
+}) => {
+  return (
+    <fieldset id="tabs">
+      <ul>
+        <li>
+          <button className="styled-button" onClick={() => updateActiveTab("chime-settings")}>
+            Preferences
+          </button>
```

Comment:
What I'd actually prefer here is to just use:

```html
<a className="tab-button" href="#chime-settings">Preferences</a>
```

And then use [onhashchange](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onhashchange) to catch the change. This way the URL reflects the location, the back button will work, you can bookmark it, etc. Also the hash should be checked on page load.

One thing I'm a little unhappy with in the current code is that `OptionsController` uses state internally, and it's not easy to get state in from the outside (or maybe I just don't understand how to do it). But adding a hashchange handler inside the function isn't good, since the function can (I believe) get called multiple times as state changes.

An example of the solution I came up with is `onKeyboardShortcutError` – it's not a great solution, but at least if you use the same solution then they will be not-great in the same way.

Reply:
Thanks @ianb I would add the mentioned changes.